ratis-netty/src/main/java/org/apache/ratis/netty/NettyUtils.java

```diff
@@ -40,4 +49,78 @@ static EventLoopGroup newEventLoopGroup(String name, int size, boolean useEpoll)
     }
     return new NioEventLoopGroup(size, ConcurrentUtils.newThreadFactory(name + "-"));
   }
+
+  static void setTrustManager(SslContextBuilder b, TrustManagerConf trustManagerConfig) {
+    if (trustManagerConfig == null) {
+      return;
+    }
+    final CertificatesConf certificates = trustManagerConfig.getTrustCertificates();
+    if (certificates.isFileBased()) {
+      b.trustManager(certificates.getFile());
+    } else {
+      b.trustManager(certificates.get());
+    }
+  }
+
+  static void setKeyManager(SslContextBuilder b, KeyManagerConf keyManagerConfig) {
+    if (keyManagerConfig == null) {
+      return;
+    }
+    final PrivateKeyConf privateKey = keyManagerConfig.getPrivateKey();
+    final CertificatesConf certificates = keyManagerConfig.getKeyCertificates();
+
+    if (keyManagerConfig.isFileBased()) {
+      b.keyManager(certificates.getFile(), privateKey.getFile());
+    } else {
+      b.keyManager(privateKey.get(), certificates.get());
```

Comment:
Unrelated to PR. It would be better to have same order of private key and certificates in the constructors for SslContextBuilder.